### DISCUSSION TITLE
User Spooled File Browser

### DISCUSSION THREAD

**Discussion by m-tyler** [codefori/vscode-ibmi#1335] [Ideas]
I am expanding my VS Code extension experience and in doing so I cobbled together a user spooled file browser based on a fork of this project. 

## New view for User Spooled Files  

1. Filters are a system user name  
1. Filters not the same name as current user are readonly
    1. No modifying of those spooled files
1. Spooled file list obtained using table function QSYS2.SPOOLED_FILE_INFO().  
1. Path scheme of `spooledfiles`  
1. Opening of spooled file into local edit tab done using CPYSPLF
    1. Opening by default opens contents into text editor tab.
    1. Conversion to PDF available but not currently working 
1. Search of spooled file contents
    1. Using right click action
    1. Using command palette action
1. Deletion of all spooled files
1. Deletion of one spooled file

## TODOS
1. Ability to delete several like named spooled files without SSH deamon error.  
1. Ability to add user profile text and spooled file quantity to filter description.  
1. Search view does not account for new path scheme of `spooledfile` for the open action.  
1. Spooled file view filter does not sort by desc date, or desc name
1. Conversion to PDF from within new `downloadSpooledFileContent` fails when this step completes `await readFileAsync(tmpobj, 'utf8')`. 
    1. Failure is that the PDF report section is encoded wrong.  I tried binary encoding but that did not work.
1. Found: Sort on members does not work by date.   

Sample images
Context menu options for current user
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/1353665/feb01581-36ac-45c0-afd1-65163b987889)

Context menu options for readonly user
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/1353665/797b8c94-059b-4ab9-8b6d-95c2e1c53e0c)

Double click on tree item to open in editor tab.
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/1353665/c5bfe1c9-fb2b-4b16-a797-a38392805d38)

Now that I have this done except for the list of TODOs is there any interest in this work and what is my next step if there is?

**Comment by worksofliam** [comment]
> Now that I have this done except for the list of TODOs is there any interest in this work and what is my next step if there is?

Best next step is to make a pull request of your fork. You should create it as a draft so you can continue your work. When you're totally finished, you can convert it the PR to a ready state, where it can be reviewed by many!

**Reply by worksofliam** [reply]
btw, I am excited. Great work!
